src/olmo_core/train/callbacks/beaker.py

```diff
@@ -0,0 +1,118 @@
+import logging
+import os
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, ClassVar, Optional
+
+from olmo_core.distributed.utils import get_rank
+from olmo_core.exceptions import OLMoEnvironmentError
+
+from .callback import Callback
+from .comet import CometCallback
+from .wandb import WandBCallback
+
+if TYPE_CHECKING:
+    from beaker import Beaker
+
+log = logging.getLogger(__name__)
+
+
+BEAKER_EXPERIMENT_ID_ENV_VAR = "BEAKER_EXPERIMENT_ID"
+
+
+@dataclass
+class BeakerCallback(Callback):
+    """
+    Adds metadata to the Beaker experiment description when running as a Beaker batch job.
+    """
+
+    priority: ClassVar[int] = min(CometCallback.priority - 1, WandBCallback.priority - 1)
+    experiment_id: Optional[str] = None
+    update_interval: Optional[int] = None
```

Comment:
I don't know enough about beaker APIs, but do we have to worry about how often it updates? I can imagine for small 1b runs, this might update multiple time per second, and maybe beaker APIs are unhappy about that.

Reply:
Good point. 4bece53